### ISSUE TITLE
only works with Java 8?

### ISSUE THREAD

**Issue by mandubian** [jove-sh/jove-spark#1]
I've tried to run it on my ArchLinux with JDK7 and I get:

```
java.lang.UnsupportedClassVersionError: jove/xsbti/ArtifactInfo : Unsupported major.minor version 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at jove.sbt.ScalaArtifacts$.<init>(IvyScala.scala:19)
	at jove.sbt.ScalaArtifacts$.<clinit>(IvyScala.scala)
	at jove.sbt.IvyScala$.$lessinit$greater$default$7(IvyScala.scala:39)
	at com.github.alexarchambault.artifact.app.IvyUtil$$anonfun$1.apply(IvyUtil.scala:14)
	at com.github.alexarchambault.artifact.app.IvyUtil$$anonfun$1.apply(IvyUtil.scala:13)
	at scala.Option.map(Option.scala:145)
	at com.github.alexarchambault.artifact.app.IvyUtil$.fakeModule(IvyUtil.scala:13)
```

**Comment by alexarchambault** [comment]
Indeed, I made the mistake of building the 0.1.0 release with Java 8 without asking scalac/javac to generate Java 7 compatible output, with a `scalacOptions += "-target:jvm-1.7"`.

That's for 0.1.0, now the development version (0.1.1-SNAPSHOT), jove-scala/jove-spark now depend on Li Haoyi's Ammonite, which requires Java 8, even to just be built. So jove-scala/jove-spark do require it too.

I'm on my way to make Ammonite optional, and/or convince Li Haoyi to make Ammonite Java 7 compatible. But in the mean time, Java 8 is required... Sorry about that.

**Comment by alexarchambault** [comment]
Alternatively, if you're willing to build it yourself, that can be done:

    $ git clone --recursive https://github.com/jove-sh/jove-meta.git
    $ cd jove-meta
    $ sbt jove-spark-cli-12/pack
    $ rm -f jove-spark/cli/target/spark-1.2/pack/lib/scala-*2.11.[45]*
    $ for i in compiler library reflect; do
        wget "https://oss.sonatype.org/content/repositories/releases/org/scala-lang/scala-$i/2.11.2/scala-$i-2.11.2.jar" -O "jove-spark/cli/target/spark-1.2/pack/lib/scala-$i-2.11.2.jar"
      done

Then run

    $ ./jove-spark/cli/target/spark-1.2/pack/bin/jove-spark-1.2

(I'm testing this right now, but that should work.)

**Comment by alexarchambault** [comment]
Actually, building it requires more work than that (you would have to checkout slightly different versions - 0.1.0-1 - of jove-scala and jove-spark, and publish the right projects of them locally). I think it would be easier for you to either find a JDK 8 or wait for Jove and/or Ammonite updates.

**Comment by mandubian** [comment]
I must switch to JDK8 anyway so it's maybe time :D

**Comment by mandubian** [comment]
BTW with JDK8, when I run `bin/jove-spark`, I've got this error:

```
Unable to launch jove-spark: unresolved dependency: sh.jove#jove-spark-cli_1.2_2.11;0.1.0-SNAPSHOT: not found
```

I haven't tried to go further, no time right now ;)

**Comment by alexarchambault** [comment]
Arggh.. Fixed this kind of things for notebook and the scala kernel, but not there. That's SNAPSHOT artifacts of previous versions disappearing after a release.

I commited fixes, and tested on a clean machine, this should now work. Pardon this so young project, thanks for the report =)

Just so that you know too:
- Ammonite-like output is not enabled when used from ipython (you'll get b&w output - one needs jove-console for colors, which is not released yet), 
- the jove-notebook command that is printed doesn't work as is yet.

If ever you try to use spark from jove-notebook (`cs jove-sh/jove-notebook; jove-notebook`), don't trust the front-end if it says that the connection is broken. It should not be, start entering commands. That's a minor regression that appeared around the release...

Thanks again!

**Comment by mandubian** [comment]
no worries, I'm just testing for now, I don't really know ipython/jupyter world too so I'm discovering at same time!

**Comment by alexarchambault** [comment]
fyi, I deprecated the conscript install. The recommended way of doing is now to use ready-to-run packages. Starts up faster, and easier to maintain. (But still no luck with JDK 7 =( )

**Comment by mandubian** [comment]
Just for my knowledge, why had you decided to use conscript (and to deprecate it now)?

**Comment by alexarchambault** [comment]
For various reasons, mainly because it didn't deal well with spark:
- in 2.10, various dependencies of spark were not handled well by launchconfig, the mechanism from sbt that conscript uses to launch apps (see https://github.com/sbt/sbt/issues/1800). I had to re-publish them under my own organization under a different name (see https://github.com/jove-sh/jove-spark/blob/master/project/JoveSparkBuild.scala#L107)
- in 2.11, there are binary incompatibilities in the scala-compiler JAR (between 2.11.2 and 2.11.4 or 5) which make the spark repl crash, and I found no way to enforce 2.11.2 (sbt treats the scala deps in a special way), so I had to bootstrap the various apps (jove-spark, jove-notebook, jove-console) to put the right JAR in the classpath. Plus setting-up / using bootstrapped apps was cumbersome.

Also, downloading all the dependencies through Ivy on a clean environment takes time, which makes it hard to test manually. Targz packages are much more straightforward to download and use.

**Comment by alexarchambault** [comment]
Do you regret the conscript install? I wish running apps from artifacts this way worked better. Maybe a possible switch of sbt to aether, or writing a launcher from scratch (which I ended up trying, https://github.com/alexarchambault/artifact-app) would make this more practical.

**Comment by mandubian** [comment]
not at all, conscript is a discovery for me (I knew the name but nothing
more :))
